blackjax/util.py

```diff
@@ -179,26 +181,64 @@ def run_inference_algorithm(
         2. The trace of states of the inference algorithm (contains the MCMC samples).
         3. The trace of the info of the inference algorithm for diagnostics.
     """
-    init_key, sample_key = split(rng_key, 2)
-    try:
-        initial_state = inference_algorithm.init(initial_state_or_position, init_key)
-    except (TypeError, ValueError, AttributeError):
-        # We assume initial_state is already in the right format.
-        initial_state = initial_state_or_position
 
-    keys = split(sample_key, num_steps)
+    keys = split(rng_key, num_steps)
 
     @jit
     def _one_step(state, xs):
         _, rng_key = xs
         state, info = inference_algorithm.step(rng_key, state)
         return state, (transform(state), info)
 
+    def _online_one_step(average_and_state, xs):
```

Comment:
There is too much code repetition here, I suggest always computing the average as additional state for now.

Reply:
Addressed.